src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs

```diff
@@ -43,6 +45,50 @@ public static Task<RemoteHostClient> TryGetRemoteHostClientAsync(this Workspace
             return clientService?.TryGetRemoteHostClientAsync(cancellationToken);
         }
 
+        public static Task<RemoteHostClient.Session> TryCreateCodeAnalysisServiceSessionAsync(
+             this Solution solution, Option<bool> option, string experimentName, CancellationToken cancellationToken)
+             => TryCreateCodeAnalysisServiceSessionAsync(solution, option, experimentName, callbackTarget: null, cancellationToken: cancellationToken);
+
+        public static bool IsOutOfProcessEnabled(this Workspace workspace, Option<bool> option, string experimentName)
```

Comment:
Will do. However, one reason it's public is that in a following PR we will be referencing this from another locatoin.   Specifically a couple of our incremental analyzers need to decide if they should actually analyze.  And the in-proc one will not analyze if all these features are actually rnning oop.